Shared/Samples/Navigate route with rerouting/NavigateRouteWithReroutingView.Model.swift

```diff
@@ -0,0 +1,304 @@
+                networkName: "Streets_ND"
+            )
+            
+            // Create the route parameters.
+            let parameters = try await routeTask.makeDefaultParameters()
+            parameters.returnsDirections = true
+            parameters.returnsStops = true
+            parameters.outputSpatialReference = .wgs84
+            
+            // Sets the start and destination stops for the route.
+            let startStop = Stop(point: .startLocation)
+            startStop.name = "San Diego Convention Center"
+            
+            let destinationStop = Stop(point: .destinationLocation)
+            destinationStop.name = "RH Fleet Aerospace Museum"
+            
+            parameters.setStops([startStop, destinationStop])
+            
+            // Solve the route using the parameters and task.
+            let routeResult = try await routeTask.solveRoute(using: parameters)
+            route = routeResult.routes.first
+            
+            // Make the route tracker.
+            routeTracker = try await makeRouteTracker(
+                routeResult: routeResult,
+                routeTask: routeTask,
+                routeParameters: parameters
+            )
+            
+            // Set up the data source's locations using a local JSON file.
+            let jsonData = try Data(contentsOf: .sanDiegoTourPath)
+            guard let jsonString = String(data: jsonData, encoding: .utf8) else { return }
+            let routePolyline = try Polyline.fromJSON(jsonString)
+            simulatedDataSource.setSimulatedLocations(with: routePolyline)
+            
+            initializeGraphics()
+        }
+        
+        /// Starts the location display.
+        func start() async throws {
+            try await locationDisplay.dataSource.start()
+            locationDisplay.autoPanMode = .navigation
+        }
+        
+        /// Stops the location display.
+        func stop() async {
+            // Stop any current speech.
+            speechSynthesizer.stopSpeaking(at: .immediate)
+            
+            // Reset the location display.
+            await locationDisplay.dataSource.stop()
+            locationDisplay.autoPanMode = .off
+        }
+        
+        /// Resets the navigation.
+        func reset() async throws {
+            await stop()
+            
+            // Reset the graphics.
+            statusMessage = .initialInstructions
+            traversedRouteGraphic.geometry = nil
+            traversedRouteBuilder.replaceGeometry(with: nil)
+            
+            // Reset the route.
+            routeTracker.cancelRerouting()
+            simulatedDataSource.currentLocationIndex = 0
+            initializeGraphics()
+        }
+        
+        /// Updates the status message and route graphics using the progress from a given tracking status.
+        /// - Parameter status: The `TrackingStatus`.
+        func updateGraphics(using status: TrackingStatus) async {
```

Comment:
It would be good if this function name indicated that more than just graphics are updated. Particularly because graphics would probably be considered just what's in the graphics overlay and this updates more than that.